ApollosProject/apollos-prototype
# Adds "Soft" Hyphenation of HorizontalHighlightCard Item Titles

## DESCRIPTION

### What does this PR do, or why is it needed?

Adds a parameter to queries which fetch content items that allows them to return the title of the item with soft hyphens placed within it.

The reason we have this at all is because titles were sometimes wrapping within horizontal cards without hyphens; it was not on.

### What design trade-offs/decisions were made?

We elected to only hyphenate words whose length were greater than 8 characters. 8 characters being the current maximum number of Ms that could fit on one line in the existing width of the horizontal cards, given the font size.

### How do I test this PR?

1. Start up an IOS sim
1. Navigate to the discover tab
2. Scroll through the content items and find one with a hyphen
3. If you want to look at Android, you should verify that nothing has changed for it.

I have some content items in the screenshots below that could work for this purpose.

### What automated tests did you write?

## SCREENSHOTS

|IOS|Android|
|---|---|
|<img width="569" alt="Screen Shot 2019-10-21 at 9 39 07 AM" src="https://user-images.githubusercontent.com/29125070/67210233-b834b380-f3e6-11e9-9f8c-1ebfaa1076ce.png">|<img width="545" alt="Screen Shot 2019-10-21 at 9 51 59 AM" src="https://user-images.githubusercontent.com/29125070/67211345-93414000-f3e8-11e9-89a1-b8aac5bcfa92.png">|


## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
